src/read.rs

```diff
@@ -226,48 +225,45 @@ impl<R> Reader<R>
     /// The format is determined by attempting to construct each specific format reader until one
     /// is successful.
     pub fn new(mut reader: R) -> Result<Self, ReadError> {
-        let mut reader_errors = Vec::new();
 
         #[cfg(feature="wav")]
         {
-            let maybe_err = match hound::WavReader::new(&mut reader) {
-                Ok(_) => None,
-                Err(err) => Some(err),
+            let is_wav = match hound::WavReader::new(&mut reader) {
+                Err(hound::Error::FormatError(_)) => false,
+                _ => true,
```

Comment:
Remember we attempt to construct the `WavReader` again immediately after this line during which any IO errors that might have occurred in this previous call would likely occur again and be propagated upwards via the `try!` macro. Do you feel this is sufficient? Or would you prefer I add another branch here for returning the IO error as soon as possible?